sqlglot/expressions.py

```diff
@@ -274,9 +365,172 @@ class CharacterSet(Expression):
+        return _apply_delegate(
+            self,
+            "limit",
+            expression=expression,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+        )
+
+    def offset(self, expression, dialect=None, parser_opts=None, copy=True):
+        return _apply_delegate(
+            self,
+            "offset",
+            expression=expression,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+        )
+
+    def lateral(
+        self, *expressions, append=True, dialect=None, parser_opts=None, copy=True
+    ):
+        return _apply_delegate(
+            self,
+            "lateral",
+            *expressions,
+            append=append,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+        )
+
+    def join(
+        self,
+        *expressions,
+        append=True,
+        dialect=None,
+        parser_opts=None,
+        copy=True,
+        prefix="JOIN",
+    ):
+        return _apply_delegate(
+            self,
+            "join",
+            *expressions,
+            append=append,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+            prefix=prefix,
+        )
+
+    def where(
+        self, *expressions, append=True, dialect=None, parser_opts=None, copy=True
+    ):
+        return _apply_delegate(
+            self,
+            "where",
+            *expressions,
+            append=append,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+        )
+
+    def having(
+        self, *expressions, append=True, dialect=None, parser_opts=None, copy=True
+    ):
+        return _apply_delegate(
+            self,
+            "having",
+            *expressions,
+            append=append,
+            dialect=dialect,
+            parser_opts=parser_opts,
+            copy=copy,
+        )
+
+    def distinct(self, distinct=True, copy=True):
+        return _apply_delegate(self, "distinct", distinct=distinct, copy=copy)
+
+    def with_(
+        self,
+        alias,
+        as_,
+        recursive=None,
+        append=True,
+        dialect=None,
+        parser_opts=None,
+        copy=True,
+    ):
+        instance = _maybe_copy(self, copy)
+        alias_expression = _maybe_parse(
+            alias,
+            dialect=dialect,
+            expression_type=TableExpressionAlias,
+            parser_opts=parser_opts,
+        )
+        as_expression = _maybe_parse(
+            as_,
+            dialect=dialect,
+            parser_opts=parser_opts,
+        )
+        expression = TableExpression(
+            this=as_expression,
+            alias=alias_expression,
+        )
+
+        expressions = [expression]
+        existing_expressions = instance.args.get("expressions")
+        if append and existing_expressions:
+            expressions = existing_expressions + expressions
```

Comment:
can we just use append here or extend?

Reply:
Ah, I see why I did it this way everywhere - I want `expressions` to be at the end of the list, but I want `expressions` to also be the resulting variable.

So it's either:

```python
expressions = existing_expressions + expressions
```
or
```python
existing_expressions.extend(expressions)
expressions = existing_expressions
```

Reply:
ok keep it then